pkg/client/apiutil/apimachinery.go

```diff
@@ -169,3 +144,23 @@ func createRestConfig(gvk schema.GroupVersionKind, isUnstructured bool, baseConf
 
 	return cfg
 }
+
+// protoMessage is implemented by protobuf messages (of all libraries)
+type protoMessage interface {
+	ProtoMessage()
+}
+
+var protoMessageType = reflect.TypeOf(new(protoMessage)).Elem()
+
+// haveProtoDefinition returns true if the go type for the specified gvk support protobuf encoding.
+func haveProtoDefinition(scheme *runtime.Scheme, gvk schema.GroupVersionKind) bool {
+	gvkType, found := scheme.AllKnownTypes()[gvk]
+	if !found {
+		return false
+	}
+	// We normally get the raw struct e.g. v1.Pod, not &v1.Pod
+	if gvkType.Kind() == reflect.Struct {
+		gvkType = reflect.PtrTo(gvkType)
+	}
+	return gvkType.Implements(protoMessageType)
```

Comment:
Is this slow enough that we should cache it? I know reflect operations can get pretty cranky sometimes.

Reply:
It did involve a linear scan, so I added memoization - thanks!